src/transformers/pipelines/visual_question_answering.py

```diff
@@ -87,29 +140,109 @@ def __call__(self, image: Union["Image.Image", str], question: str = None, **kwa
+    def preprocess(self, inputs, padding=False, truncation=False, words=[], boxes=[]):
+        image_features = {}
+        if "image" in inputs:
+            image = load_image(inputs["image"])
+            image_features = self.feature_extractor(images=image, return_tensors=self.framework)
+
+        # TODO: If the image is specified, depending on the tokenizer, we should update inputs
+        # with the words/bounding boxes.
+        #
+        # TODO: LayoutLMv1 does not come with a feature extractor that can run OCR, but v2 and v3 do.
+        # We could either (a) require users of v1 pass in words (or implement a v1 feature extractor)
+        # or (b) build the OCR implementation into this pipeline (and expect users of v2 and v3 to
+        # instantiate the pipeline with feature extractors that have OCR disabled).
+
+        if not ("image" in inputs or "words" in inputs):
+            raise ValueError("Must provide at least one of an image or words/bounding boxes")
+
+        question = inputs["question"]
+        text_pair = None
+
+        extra_tokenizer_params = {}
+        if "words" in inputs:
+            # TODO: Can we refactor and share (or inherit from) the QuestionAnweringPipeline? I think we want to do
+            # the same thing with the tokenizer, except we have a few extra arguments (e.g. the bounding boxes).
+            extra_tokenizer_params = {
+                "return_token_type_ids": True,
+                "return_attention_mask": True,
+                "is_split_into_words": True,
+                # "max_length": 512,
+            }
+            padding = "max_length"
+            truncation = True
+            question = [question]
+            text_pair = inputs["words"]
+
+        encoding = self.tokenizer(
+            text=question,
+            text_pair=text_pair,
+            return_tensors=self.framework,
+            padding=padding,
+            truncation=truncation,
+            **extra_tokenizer_params,
         )
-        image_features = self.feature_extractor(images=image, return_tensors=self.framework)
-        model_inputs.update(image_features)
-        return model_inputs
 
-    def _forward(self, model_inputs):
-        model_outputs = self.model(**model_inputs)
-        return model_outputs
-
-    def postprocess(self, model_outputs, top_k=5):
-        if top_k > self.model.config.num_labels:
-            top_k = self.model.config.num_labels
-
-        if self.framework == "pt":
-            probs = model_outputs.logits.sigmoid()[0]
-            scores, ids = probs.topk(top_k)
+        num_spans = len(encoding["input_ids"])
+
+        if "boxes" in inputs:
+            boxes = inputs["boxes"]
+            bbox = []
+            for batch_index in range(num_spans):
+                for i, s, w in zip(
+                    encoding.input_ids[batch_index],
+                    encoding.sequence_ids(batch_index),
+                    encoding.word_ids(batch_index),
```

Comment:
Yes. You could read this code colloquially as "if the box belongs to a word in the doc (i.e. sequence id is 1) then assign it a bounding box based on its original (word id) index. If the box belongs to a separator token, assign it [1000,1000,1000,1000]. Otherwise, assign it to [0,0,0,0]".

The challenge you're raising (words vs. boxes in the image) is a very real challenge for document processing tools. Most do not support "within-box" predictions. Definitely a problem worth visiting in the future.